src/v0/destinations/pinterest_tag/transform.js

```diff
@@ -164,15 +152,17 @@ const processRouterDest = (inputs, reqMetadata) => {
     }
   });
 
-  const batchResponseList = [];
+  // using the first destination config for transforming the batch
+  const { destination } = inputs[0];
+  const { apiVersion, adAccountId, conversionToken } = destination.Config;
+  const endpoint = getEventsEndpoint(adAccountId);
+  let batchResponseList = [];
   if (successRespList.length > 0) {
-    const batchedEvents = batchMultiplexedEvents(successRespList, MAX_BATCH_SIZE);
-    batchedEvents.forEach((batch) => {
-      const batchedRequest = generateBatchedPayloadForArray(batch.events);
-      batchResponseList.push(
-        getSuccessRespEvents(batchedRequest, batch.metadata, batch.destination, true),
-      );
-    });
+    if (apiVersion === API_VERSION.v5) {
+      batchResponseList = batchEventsNew(successRespList, endpoint, conversionToken);
```

Comment:
same for `batchEvents` and `batchEventsNew`, both are same, except for authorization. Let's consider it an individual function that works for v3 and v5. Instead, we can create a separate util function for generating the header section of the request, which seems to be different.